crates/biome_js_type_info/src/type_info.rs

```diff
@@ -1106,24 +1121,71 @@ impl TypeReference {
                 .iter()
                 .map(|param| param.resolved(resolver))
                 .collect(),
-            Self::Imported(import) => import
-                .type_parameters
-                .iter()
-                .map(|param| param.resolved(resolver))
-                .collect(),
             _ => [].into(),
         }
     }
 }
 
-/// Imported identifier, with associated type parameters.
+/// Qualifier for a type that should be imported from another module.
 #[derive(Clone, Debug, PartialEq)]
 pub struct TypeImportQualifier {
-    /// Name of the imported symbol.
-    pub identifier: Text,
+    /// The imported symbol.
+    pub symbol: ImportSymbol,
 
-    /// Generic type parameters specified in the reference.
-    pub type_parameters: Box<[TypeReference]>,
+    /// Resolved path of the module to import the type from.
+    pub resolved_path: ResolvedPath,
+}
+
+/// Reference-counted resolved path wrapped in a [Result] that contains a string
+/// message if resolution failed.
+#[derive(Clone, Debug, PartialEq)]
+pub struct ResolvedPath(Arc<Result<Utf8PathBuf, String>>);
+
+impl Deref for ResolvedPath {
+    type Target = Result<Utf8PathBuf, String>;
+
+    fn deref(&self) -> &Self::Target {
+        self.0.as_ref()
+    }
+}
+
+impl ResolvedPath {
+    pub fn new(resolved_path: Result<Utf8PathBuf, String>) -> Self {
+        Self(Arc::new(resolved_path))
+    }
+
+    pub fn as_path(&self) -> Option<&Utf8Path> {
+        self.as_deref().ok()
+    }
+
+    pub fn from_path(path: impl Into<Utf8PathBuf>) -> Self {
+        Self::new(Ok(path.into()))
+    }
+}
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub enum ImportSymbol {
+    /// Imports the `default` export.
+    #[default]
+    Default,
```

Comment:
It feels hacky to have `Default` as the default implementation, because we might don't know if it's actually the import default, or we forgot to update the symbol. 

What if we had a `_None` variant, and then having a `debug_assert!` that checks its value. If `_None` is encountered, it means that the type hasn't been updated somewhere

Reply:
Looks like we don't need a `Default` for it at all :)